modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/util/v1_0/SkuOptionUtil.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.commerce.delivery.catalog.internal.util.v1_0;
+
+import com.liferay.commerce.product.model.CPDefinitionOptionRel;
+import com.liferay.commerce.product.model.CPDefinitionOptionValueRel;
+import com.liferay.headless.commerce.delivery.catalog.dto.v1_0.SkuOption;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Crescenzo Rega
+ */
+public class SkuOptionUtil {
+
+	public static SkuOption[] getSkuOptions(
+			Map<CPDefinitionOptionRel, List<CPDefinitionOptionValueRel>>
+				cpDefinitionOptionRelsMap,
+			String languageId)
+		throws Exception {
+
+		List<SkuOption> skuOptions = new ArrayList<>();
+
+		for (Map.Entry<CPDefinitionOptionRel, List<CPDefinitionOptionValueRel>>
+				entry : cpDefinitionOptionRelsMap.entrySet()) {
+
+			CPDefinitionOptionRel cpDefinitionOptionRel = entry.getKey();
+
+			List<CPDefinitionOptionValueRel> cpDefinitionOptionValueRels =
+				entry.getValue();
+
+			for (CPDefinitionOptionValueRel cpDefinitionOptionValueRel :
+					cpDefinitionOptionValueRels) {
+
+				SkuOption skuOption = new SkuOption() {
+					{
+						id = cpDefinitionOptionRel.getCPDefinitionOptionRelId();
+						key = cpDefinitionOptionRel.getKey();
+						skuOptionValueId =
+							cpDefinitionOptionValueRel.
+								getCPDefinitionOptionValueRelId();
+						skuOptionValueKey = cpDefinitionOptionValueRel.getKey();
+						value = String.valueOf(
+							cpDefinitionOptionValueRel.
```

Comment:
```suggestion
						id = cpDefinitionOptionRel.getCPDefinitionOptionRelId();
						key = cpDefinitionOptionRel.getKey();
						skuOptionValueKey = cpDefinitionOptionValueRel.getKey();
						value = String.valueOf(
							cpDefinitionOptionValueRel.
						valueId =
							cpDefinitionOptionValueRel.
								getCPDefinitionOptionValueRelId();
```

Reply:
In my opinion one of these two for consistency, plus I don't understand why you need to also add the toString of the value = String.valueOf(cpDefinitionOptionValueRel.) and if you need this why you didn't add the toString of the optionId?

Reply:
Before the changes, the value field was of integer type and was set to `cpDefinitionOptionValueRel.getCPDefinitionOptionValueRelId()` we changed it to string type but to maintain backwards compatibility we must continue to set the field to the value of `cpDefinitionOptionValueRel.getCPDefinitionOptionValueRelId()`